qiskit_experiments/library/characterization/resonator_spectroscopy.py

```diff
@@ -0,0 +1,218 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Spectroscopy experiment class for resonators."""
+
+from typing import Iterable, Optional, Tuple
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.exceptions import QiskitError
+from qiskit.providers import Backend
+import qiskit.pulse as pulse
+
+from qiskit_experiments.framework import Options
+from qiskit_experiments.library.characterization.spectroscopy import Spectroscopy
+from .analysis.resonator_spectroscopy_analysis import ResonatorSpectroscopyAnalysis
+
+
+class ResonatorSpectroscopy(Spectroscopy):
```

Comment:
Since this experiment does not depend on the qubit, it could run at a high rep rate without qubit reset. I don't think is a good way to set a high rep rate and disable reset from the experiment? You would need to do some logic to determine what run options to set. It might be nice to show setting the options for a high rep rate and no reset in use in a tutorial.

Reply:
You can use `#section: examples` to write some code. As you mentioned, we cannot set default value to run option since this is provider dependent. Even if we don't assume non-IBM provider for experiments, simulator may crash with invalid options (probably meas level and meas return are okey since they are defined in OpenPulse spec).